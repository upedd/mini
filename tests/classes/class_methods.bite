class Foo {
    get10() {
        return 10;
    }

    add(a, b) {
        return a + b;
    }

    add_to_10(a) {
        return this.add(a, this.get_10());
    }
}
native print;

let instance = Foo();

print(instance.get10());
print(instance.add(3, 5));
print(instance.add_to_10(7));

let instance2 = Foo();
print(instance2.get10());
print(instance2.add(4, 3));
print(instance2.add_to_10(2));