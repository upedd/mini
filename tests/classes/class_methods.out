10
8
17
10
7
12